pkgs/development/compilers/opensycl/default.nix

```
@@ -0,0 +1,49 @@
+{ lib, clang15Stdenv, fetchFromGitHub, llvmPackages_15, lld_15, llvmPackages_rocm, rocm-device-libs, python3, rocm-runtime, cmake, boost, libxml2, libffi, makeWrapper, hip, rocmSupport ? false }:
+
+clang15Stdenv.mkDerivation {
+  pname = "OpenSYCL";
+  version = "0.9.4";
+  src = fetchFromGitHub {
+    owner = "OpenSYCL";
+    repo = "OpenSYCL";
+    rev = "v0.9.4";
+    sha256 = "sha256-5YkuUOAnvoAD5xDKxKMPq0B7+1pb6hVisPAhs0Za1ls=";
+  };
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+  ];
+  buildInputs = [
+    libxml2
+    libffi
+    boost
+    llvmPackages_15.openmp
+    llvmPackages_15.libclang.dev
+    llvmPackages_15.llvm
+  ] ++ lib.optionals rocmSupport [
+    hip
+    rocm-runtime
+  ];
```

```suggestion
clang15Stdenv.mkDerivation {
  pname = "OpenSYCL";
  version = "0.9.4";

  src = fetchFromGitHub {
    owner = "OpenSYCL";
    repo = "OpenSYCL";
    rev = "v0.9.4";
    sha256 = "sha256-5YkuUOAnvoAD5xDKxKMPq0B7+1pb6hVisPAhs0Za1ls=";
  };

  nativeBuildInputs = [
    cmake
    makeWrapper
  ];

  buildInputs = [
    libxml2
    libffi
    boost
    llvmPackages_15.openmp
    llvmPackages_15.libclang.dev
    llvmPackages_15.llvm
  ] ++ lib.optionals rocmSupport [
    hip
    rocm-runtime
  ];
```

just a small formatting nitpick

---

Done